examples/exNDH1.cpp

```diff
@@ -0,0 +1,311 @@
+//                       MFEM Example 3 - Parallel Version
+//
+// Compile with: make exNDH1
```

Comment:
I agree that this is an important test.  I think we can make the case that this should be an example if we add (or at least promise to add) code demonstrating a wider variety of spatial derivatives.  First derivatives, like this gradient, just don't appear in that many examples so users often ask about them.

At the very least I think we should add the discrete interpolator which computes the gradient of an H1 field since, in many cases, this is sufficient and simpler than solving for the result in H(curl).  It would also provide further validation of the PA operator.

Ideally we could write this in a way analogous to `ex22` and support a variety of field types.  We could then demonstrate; Grad of H1, Curl of H(curl), Div of H(div), Weak Grad of L2, Weak Curl of H(div), and Weak Div of H(curl).  Even further in the future we could add more exotic first derivatives like directional derivatives.

If this sounds reasonable, let's add the `GradientInterpolator` now and add something to the comment block at the top indicating future plans (in case anyone questions the value of this example).

Reply:
This sounds good to me, and it could become a regression test. @tzanio do you agree to adding this as an example? If so, I will add a GradientInterpolator test in this example for comparison, and make a serial version of the example.

Reply:
Yes, especially if it does all the strong and weak derivatives that Mark mentioned.

Reply:
For this PR, just (Nedelec, gradient H1). The others will be good to add in the future.